src/config/network.config.ts

```diff
@@ -0,0 +1,57 @@
+import { Network } from "@/utils/wallet/wallet_provider";
+
+const network = process.env.NEXT_PUBLIC_NETWORK as Network;
+
+interface NetworkConfig {
+  coinName: string;
+  coinSymbol: string;
+  mempoolApiUrl: string;
+}
+
+const mainnetConfig: NetworkConfig = {
+  coinName: "BTC",
+  coinSymbol: "BTC",
+  mempoolApiUrl: `${process.env.NEXT_PUBLIC_MEMPOOL_API}`,
+};
+
+const signetConfig: NetworkConfig = {
+  coinName: "Signet BTC",
+  coinSymbol: "sBTC",
+  mempoolApiUrl: `${process.env.NEXT_PUBLIC_MEMPOOL_API}/signet`,
+};
+
+const config: Record<string, NetworkConfig> = {
+  mainnet: mainnetConfig,
+  testnet: signetConfig,
+};
+
+export function getNetworkConfig(): NetworkConfig {
+  switch (network) {
+    case Network.MAINNET:
+      return config.mainnet;
+    case Network.REGTEST:
+    case Network.SIGNET:
+    case Network.TESTNET:
+      return config.testnet;
```

Comment:
```suggestion
const signetConfig: NetworkConfig = {
  coinName: "Signet BTC",
  coinSymbol: "sBTC",
  mempoolApiUrl: `${process.env.NEXT_PUBLIC_MEMPOOL_API}/signet`,
};

const testnetConfig: NetworkConfig = {
  coinName: "Testnet BTC",
  coinSymbol: "tBTC",
  mempoolApiUrl: `${process.env.NEXT_PUBLIC_MEMPOOL_API}/testnet`,
};

const config: Record<string, NetworkConfig> = {
  mainnet: mainnetConfig,
  signet: signetConfig
  testnet: testnetConfig,
};

export function getNetworkConfig(): NetworkConfig {
  switch (network) {
    case Network.MAINNET:
      return config.mainnet;
    case Network.SIGNET:
      return config.signet;
    case Network.TESTNET:
      return config.testnet;
```

Couple things:
- Let's not support regtest as wallets can't connect to it and there's no mempool API for it.
- Testnet and signet are different networks